sdk/src/main/java/me/uport/sdk/Transactions.kt

```diff
@@ -119,25 +121,28 @@ class Transactions(
             val txHash = when (signerType) {
                 MetaIdentityManager -> {
 
-                    val metaSigner = MetaIdentitySigner(relaySigner, account.publicAddress, account.identityManagerAddress)
+                    @Suppress("UnsafeCast")
+                    val metaAccount = account as MetaIdentityAccount
+                    val metaSigner = MetaIdentitySigner(relaySigner, metaAccount.publicAddress, metaAccount.identityManagerAddress)
                     signedEncodedTx = metaSigner.signRawTx(oldBundle.unsigned)
 
                     relayMetaTransaction(signedEncodedTx)
 
                 }
-                KeyPair -> {
+                KeyPair, HDKeyPair -> {
                     signedEncodedTx = signer.signRawTx(oldBundle.unsigned)
                     relayRawTransaction(signedEncodedTx)
                 }
                 else -> {
 
                     signedEncodedTx = relaySigner.signRawTx(oldBundle.unsigned)
 
-                    if (signerType != KeyPair) {
+                    /*if (signerType != KeyPair || signerType != HDKeyPair) {
```

Comment:
this block could be deleted entirely, and replaced with a comment about the type of exception thrown when the `deviceAddress` has no funds.

It should be easy to check by trying to send a transaction with a fresh account.